indra/newview/llpanelobjectinventory.cpp

```diff
@@ -307,20 +307,27 @@ BOOL LLTaskInvFVBridge::isItemRenameable() const
 BOOL LLTaskInvFVBridge::renameItem(const std::string& new_name)
 {
     LLViewerObject* object = gObjectList.findObject(mPanel->getTaskUUID());
-    if(object)
+    if(!object || !object->permModify())
     {
-        LLViewerInventoryItem* item = NULL;
-        item = (LLViewerInventoryItem*)object->getInventoryObject(mUUID);
-        if(item && (gAgent.allowOperation(PERM_MODIFY, item->getPermissions(),
-                                        GP_OBJECT_MANIPULATE, GOD_LIKE)))
-        {
-            LLPointer<LLViewerInventoryItem> new_item = new LLViewerInventoryItem(item);
-            new_item->rename(new_name);
-            object->updateInventory(
-                new_item,
-                TASK_INVENTORY_ITEM_KEY,
-                false);
-        }
+        return false;
+    }
+    if (!object->permModify())
+    {
+        LLNotificationsUtil::add("CantModifyContentInNoModTask");
```

Comment:
We won't get this notification because of the previous check.

Reply:
Missed it, thanks...